serde-oracle-jdbc-json/src/main/java/io/micronaut/serde/oracle/jdbc/json/serde/OracleJsonBinarySerde.java

```diff
@@ -41,31 +38,16 @@ public class OracleJsonBinarySerde extends AbstractOracleJsonSerde<byte[]> {
     @Override
     @NonNull
     protected byte[] doDeserializeNonNull(@NonNull OracleJdbcJsonParserDecoder decoder, @NonNull DecoderContext decoderContext,
-                                          @NonNull Argument<? super byte[]> type) throws IOException {
-
-        OracleType.Type t = type.getAnnotationMetadata().enumValue(OracleType.class, OracleType.Type.class).orElse(null);
-        if (t == OracleType.Type.BINARY) {
-            return decoder.decodeBinary();
-        } else {
-            return getDefault().deserializeNonNull(decoder, decoderContext, type);
-        }
+                                          @NonNull Argument<? super byte[]> type) {
+        return decoder.decodeBinary();
     }
 
     @Override
     protected void doSerializeNonNull(@NonNull OracleJdbcJsonGeneratorEncoder encoder, @NonNull EncoderContext context,
-                                      @NonNull Argument<? extends byte[]> type, @NonNull byte[] value) throws IOException {
-        OracleType.Type t = type.getAnnotationMetadata().enumValue(OracleType.class, OracleType.Type.class).orElse(null);
-        if (t == OracleType.Type.BINARY) {
-            encoder.encodeValue(new OracleJsonBinaryImpl(value, false));
-        } else {
-            getDefault().serialize(
-                encoder,
-                context,
-                type,
-                value
-            );
-        }
-
+                                      @NonNull Argument<? extends byte[]> type, @NonNull byte[] value) {
+        // Expects to be base16 encoded when writing to the db
+        String strValue = OracleJsonBinaryImpl.getString(value, false);
```

Comment:
We reuse logic from AbstractOracleJsonSerde and if oracle decoder not present then we can use default one, and this is in each oracle serde implementations.